### PR TITLE
Clean up logs

### PR DESCRIPTION
**Pull request by toanngosy**
Using os.path.basename() to get the file name is inconsistent across platforms.

In Windows, file paths typically use backslashes (\\) as the path separator, whereas in Python, the os.path module treats both forward slashes (/) and backslashes (\\) as valid path separators.

However, when using os.path.basename() on Windows paths, you might encounter issues if the path uses backslashes and you're running the code in a Unix-like environment, such as macOS or Linux. In such cases, os.path.basename() might not correctly handle the backslashes, leading to unexpected results.

To ensure platform compatibility when dealing with Windows paths, it's recommended to use pathlib instead, as it provides a more consistent and platform-independent approach.

## PR Self Evaluation

- [x] I have reviewed my code to check that it contains no sensitive information.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests or modified existing tests that prove my fix is effective or that my feature works
- [x] Existing unit tests pass locally with my changes

### CODE REVIEW

File: processing/join_site_data.py
Comment (dschristianson) [inline processing/join_site_data.py:29]: Add spaces around "/" here and elsewhere in module.

**Review by dschristianson — COMMENTED** [review]
Looking good, @toanngosy! 
A few minor comments / questions.

File: processing/base_creator.py
Comment (toanngosy) [reply processing/base_creator.py:270]: I checked this again. `base_path` is actually a Path object. It is created from the function `get_BASE_attrs()`, which I change to a Path object. This result then passes to `candidate_iterator()` in the `driver`

**Review by dschristianson — APPROVED** [review]
Thanks, @toanngosy!
@yocheah should have a look. If you approve, @yocheah, don't merge yet. I need to have a think about merge / deployment plan. Thanks!

**Comment by toanngosy** [comment]
Thanks @dschristianson !

**Pull request merged**
